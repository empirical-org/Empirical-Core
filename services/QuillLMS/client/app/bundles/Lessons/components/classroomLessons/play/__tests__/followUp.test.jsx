import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import FollowUp from '../followUp';

import {
    NO_PRACTICE, PRACTICE_LATER, PRACTICE_NOW, SMALL_GROUP_AND_INDEPENDENT
} from '../../../constants';

describe('FollowUp component', () => {

  describe('when followUpOption is SMALL_GROUP_AND_INDEPENDENT', () => {
    describe('when student is flagged', () => {
      const wrapper = mount(<FollowUp followUpOption={SMALL_GROUP_AND_INDEPENDENT} followUpUrl="" isFlagged={true} /> )

      it('renders', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })

    describe('when student is not flagged', () => {
      const wrapper = mount(<FollowUp followUpOption={SMALL_GROUP_AND_INDEPENDENT} followUpUrl="" isFlagged={false} /> )

      it('renders', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })
  })

  describe('when followUpOption is PRACTICE_NOW', () => {
    const wrapper = mount(<FollowUp followUpOption={PRACTICE_NOW} followUpUrl="" /> )

    it('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('when followUpOption is PRACTICE_LATER', () => {
    const wrapper = mount(<FollowUp followUpOption={PRACTICE_LATER} followUpUrl="" /> )

    it('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('when followUpOption is NO_PRACTICE', () => {
    const wrapper = mount(<FollowUp followUpOption={NO_PRACTICE} followUpUrl="" /> )

    it('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

})
