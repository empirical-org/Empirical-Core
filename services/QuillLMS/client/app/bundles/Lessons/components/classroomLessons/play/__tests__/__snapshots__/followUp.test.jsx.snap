// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowUp component when followUpOption is NO_PRACTICE renders 1`] = `
<WatchTeacher
  followUpOption="No Follow Up Practice"
  followUpUrl=""
>
  <div
    className="follow-up-container full-page-modal-container"
  >
    <div
      className="follow-up full-page-modal"
    >
      <img
        alt="An illustration of a teacher pointing to a smart board"
        src="undefined/images/illustrations/celebrating-activity-completion.svg"
      />
      <h1>
        You have completed the lesson!
      </h1>
      <a
        className="quill-button-archived large outlined secondary focus-on-dark"
        href="undefined"
      >
        Exit lesson
      </a>
    </div>
  </div>
</WatchTeacher>
`;

exports[`FollowUp component when followUpOption is PRACTICE_LATER renders 1`] = `
<WatchTeacher
  followUpOption="All Students Practice Later"
  followUpUrl=""
>
  <div
    className="follow-up-container full-page-modal-container"
  >
    <div
      className="follow-up full-page-modal"
    >
      <img
        alt="An illustration of a teacher pointing to a smart board"
        src="undefined/images/illustrations/celebrating-activity-completion.svg"
      />
      <h1>
        You have completed the lesson!
      </h1>
      <h2>
        Your teacher has assigned some practice for later. You'll see it on your dashboard when you exit this lesson.
      </h2>
      <a
        className="quill-button-archived large outlined secondary focus-on-dark"
        href="undefined"
      >
        Exit lesson
      </a>
    </div>
  </div>
</WatchTeacher>
`;

exports[`FollowUp component when followUpOption is PRACTICE_NOW renders 1`] = `
<WatchTeacher
  followUpOption="All Students Practice Now"
  followUpUrl=""
>
  <div
    className="follow-up-container full-page-modal-container"
  >
    <div
      className="follow-up full-page-modal"
    >
      <img
        alt="An illustration of a teacher pointing to a smart board"
        src="undefined/images/illustrations/celebrating-activity-completion.svg"
      />
      <h1>
        You have completed the lesson!
      </h1>
      <h2>
        Now it's time to practice what you learned.
      </h2>
      <a
        className="quill-button-archived large outlined secondary focus-on-dark"
        href=""
      >
        Begin
      </a>
    </div>
  </div>
</WatchTeacher>
`;

exports[`FollowUp component when followUpOption is SMALL_GROUP_AND_INDEPENDENT when student is flagged renders 1`] = `
<WatchTeacher
  followUpOption="Small Group Instruction and Independent Practice"
  followUpUrl=""
  isFlagged={true}
>
  <div
    className="follow-up-container full-page-modal-container"
  >
    <div
      className="follow-up full-page-modal"
    >
      <img
        alt="An illustration of a teacher pointing to a smart board"
        src="undefined/images/illustrations/celebrating-activity-completion.svg"
      />
      <h1>
        You have completed the lesson!
      </h1>
      <h2>
        Wait for instructions from your teacher.
      </h2>
    </div>
  </div>
</WatchTeacher>
`;

exports[`FollowUp component when followUpOption is SMALL_GROUP_AND_INDEPENDENT when student is not flagged renders 1`] = `
<WatchTeacher
  followUpOption="Small Group Instruction and Independent Practice"
  followUpUrl=""
  isFlagged={false}
>
  <div
    className="follow-up-container full-page-modal-container"
  >
    <div
      className="follow-up full-page-modal"
    >
      <img
        alt="An illustration of a teacher pointing to a smart board"
        src="undefined/images/illustrations/celebrating-activity-completion.svg"
      />
      <h1>
        You have completed the lesson!
      </h1>
      <h2>
        Now it's time to practice what you learned.
      </h2>
      <a
        className="quill-button-archived large outlined secondary focus-on-dark"
        href=""
      >
        Begin
      </a>
    </div>
  </div>
</WatchTeacher>
`;
